bot.go

```diff
@@ -198,10 +207,11 @@ func (b *Bot) Start() {
 	}
 
 	// do nothing if called twice
-	if b.stopClient != nil {
+	if b.getChanStopClient() != nil {
```

Comment:
We're still racing here, it would be better to encapsulate nil checking and creating into a one mutex call. Same with the stop.